xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -2065,6 +2062,11 @@ void CVideoPlayer::HandlePlaySpeed()
 
       if (!m_State.streamsReady)
       {
+        if (m_playerOptions.fullscreen)
+        {
+          CApplicationMessenger::GetInstance().PostMsg(TMSG_SWITCHTOFULLSCREEN);
```

Comment:
isn't the switch to fullscreen pure application logic? Not sure if VP should know anything about fullscreen or not. I think it would be clearer separation of domains if this would fired from CAapp on the "OnAVStarted" message instead?